tests/integration/test_span_links.py

```diff
@@ -0,0 +1,150 @@
+import uuid
+
+from sentry_sdk.envelope import Envelope, Item, PayloadRef
+
+
+def test_event_with_span_link_in_transaction(relay, mini_sentry):
+    """Assert that a span link in the trace context is parsed and has the expected output."""
+
+    mini_sentry.add_full_project_config(42)
+    relay = relay(mini_sentry)
+
+    event_id = uuid.uuid1().hex
+    error_payload = {
+        "event_id": event_id,
+        "transaction": "/users",
+        "contexts": {
+            "trace": {
+                "trace_id": "4c79f60c11214eb38604f4ae0781bfb2",
+                "span_id": "fa90fdead5f74052",
+                "op": "navigation",
+                "links": [
+                    {
+                        "trace_id": "3c79f60c11214eb38604f4ae0781bfb2",
+                        "span_id": "ea90fdead5f74052",
+                        "sampled": True,
+                        "attributes": {"sentry.link.type": "previous_trace"},
+                    }
+                ],
+            }
+        },
+        "type": "transaction",
+        "environment": "production",
+        "release": "foo@1.2.3",
+    }
+
+    envelope = Envelope(headers={"event_id": event_id})
+    envelope.add_item(Item(PayloadRef(json=error_payload), type="event"))
+
+    relay.send_envelope(42, envelope)
+
+    envelope = mini_sentry.captured_events.get(timeout=1)
+    assert envelope
+    assert envelope.items[0].payload.json["contexts"] == {
+        "trace": {
+            "trace_id": "4c79f60c11214eb38604f4ae0781bfb2",
+            "span_id": "fa90fdead5f74052",
+            "op": "navigation",
+            "links": [
+                {
+                    "trace_id": "3c79f60c11214eb38604f4ae0781bfb2",
+                    "span_id": "ea90fdead5f74052",
+                    "sampled": True,
+                    "attributes": {"sentry.link.type": "previous_trace"},
+                }
+            ],
+            "status": "unknown",
+            "type": "trace",
+        }
+    }
+    assert mini_sentry.captured_events.empty()
+
+
+def test_event_with_span_links_in_child_span(relay, mini_sentry):
```

Comment:
nit: You could probably merge these tests into one by having a transaction with both trace context links _and_ child span links.

Reply:
Sure thing: [47b32d8](https://github.com/getsentry/relay/pull/4486/commits/47b32d8b8342762b385f842ca9fe136c930ba8f7)